Update filter feature (#166)
AY2324S1-CS2103T-F11-2/tp: docs/UserGuide.md

><br>
>- Parameters in square brackets are optional. However, you must include at least one parameter to filter by.
   ><br>
>- For all parameters aside from NAME and RSVP_STATUS, providing an empty value will filter for guests with unfilled values for that parameter.
   ><br>
>- Providing an empty `r/` value would filter for guests with unknown RSVP status.
   ><br>
>- Providing an empty `n/` value would filter for guests without a name thus returning an empty guest list. (All guests need a name)

>    ><br>

Reviewer: Ideally an empty `n/` should throw an error because there is no point filtering for an empty name. If the implementation for that is difficult, perhaps we can stick with this!

Author: Resolved!

>- The filter command will only return guests that matches all the input values across different parameters.
   ><br>
>- DIETARY_REQUIREMENT and TAG parameters can be inputted multiple times (e.g. "guest filter d/no beef d/no pork"). 
   However, do take note that these two parameters are filtered via a case-insensitive exact match (i.e. a guest with the tag "friends" would not be a valid result for the input "t/friend").
   ><br>

> >- TABLE_NUMBER and RSVP_STATUS parameter are also filtered via a case-insensitive exact match (i.e. "unknown" would not be a valid result for the input "guest filter r/no").

Reviewer: Perhaps the example can be more precise?

```suggestion
>- TABLE_NUMBER and RSVP_STATUS parameter are also filtered via a case-insensitive exact match (i.e. `guest filter r/n` will not return guests with rsvp status `no`)
```

   ><br>

> >- NAME parameter cannot be filtered using empty values.
>    ><br>
> >- Providing an empty `n/` value would filter for vendors without a name thus returning an empty vendor list. (All vendors need a name)
>    ><br>

Reviewer: I think the top two lines here should be cut off?

```suggestion
>- Providing an empty `n/` value would filter for vendors without a name thus returning an empty vendor list. (All vendors need a name)
   ><br>
```
